medical_insurance/models/medical_insurance_company.py

```diff
@@ -1,40 +1,18 @@
 # -*- coding: utf-8 -*-
-##############################################################################
-#
-#    Author: Dave Lasley <dave@laslabs.com>
-#    Copyright: 2015 LasLabs, Inc [https://laslabs.com]
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as
-#    published by the Free Software Foundation, either version 3 of the
-#    License, or (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2015-TODAY LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
 
-from openerp import fields, models, api
+from odoo import api, models
 
 
 class MedicalInsuranceCompany(models.Model):
     _name = 'medical.insurance.company'
     _description = 'Medical Insurance Providers'
-    _inherits = {'res.partner': 'partner_id', }
-    partner_id = fields.Many2one(
-        string='Related Partner',
-        comodel_name='res.partner',
-        required=True,
-        ondelete='cascade',
-    )
+    _inherit = 'medical.abstract.entity'
 
     @api.model
-    @api.returns('self', lambda value: value.id)
-    def create(self, vals):
-        vals['is_insurance_company'] = True
-        return super(MedicalInsuranceCompany, self).create(vals)
+    def _create_vals(self, vals):
```

Comment:
It would be good to include a unit test for this overload, as well as the one in the ``medical.insurance.template`` model.

Reply:
Not gonna bother here, this module's scrap for v11